lib/svg-sprite/layouter.js

```diff
@@ -10,116 +10,115 @@
+        prefix: '.svg-%s',
+        dimensions: '-dims',
+        sprite: 'svg/sprite.stack.svg',
+        example: false,
+        bust: false
     }
-},
-defaultVariables                    = {
-    date                            : (new Date()).toGMTString(),
-    invert                          : function() {
-        return function(num, render) {
-            return -parseFloat(render(num));
-        };
+};
+const defaultVariables = {
+    date: new Date().toGMTString(),
+    invert() {
+        return (num, render) => -Number.parseFloat(render(num));
     },
-    classname                       : function() {
-        return function(str, render) {
-        	var classname			= render(str).replace(/\s+/g, ' ').split(' ').pop();
-        	return classname.startsWith('.') ? classname.substr(1) : classname;
+    classname() {
+        return (str, render) => {
+            const classname = render(str).replace(/\s+/g, ' ').split(' ').pop();
+            return classname.startsWith('.') ? classname.substr(1) : classname;
         };
     },
-    escape                          : function() {
-        return function(str, render) {
-            return render(str).split('\\').join('\\\\');
-        };
+    escape() {
+        return (str, render) => render(str).split('\\').join('\\\\');
     },
-    encodeHashSign                  : function() {
-        return function(str, render) {
-            return render(str).split('#').join('%23');
-        };
+    encodeHashSign() {
+        return (str, render) => render(str).split('#').join('%23');
     }
 };
 
 /**
  * SVGSprite layouter
  *
- * @param {SVGSpriter} spriter		SVG spriter
- * @param {Object} config			Layout configuration
+ * @param {SVGSpriter} spriter      SVG spriter
+ * @param {Object} config           Layout configuration
  */
 function SVGSpriteLayouter(spriter, config) {
-    this._spriter                   = spriter;
-    this.config                     = config;
-    this.mode                       = null;
-    this.files                      = {};
-    this.data                       = {};
-    this._commonData                = Object.assign({shapes: []}, defaultVariables, this._spriter.config.variables);
+    this._spriter = spriter;
+    this.config = config;
+    this.mode = null;
+    this.files = {};
+    this.data = {};
+    this._commonData = {
+        shapes: [],
+        ...defaultVariables,
+        ...this._spriter.config.variables
+    };
 
     // Register the common shapes data
-    var shapesLength = this._spriter._shapes.length;
-    this._spriter._shapes.forEach(function(shape, index) {
-        var dimensions              = shape.getDimensions(),
-        padding                     = shape.config.spacing.padding;
+    const shapesLength = this._spriter._shapes.length;
+    this._spriter._shapes.forEach((shape, index) => {
+        const { width, height } = shape.getDimensions();
+        const { top, right, bottom, left } = shape.config.spacing.padding;
```

Comment:
@jkphl I think this might not be 100% right. Isn't `padding` an Array? If so, shouldn't we use this (untested)?

```js
const [top, right, bottom, left] = shape.config.spacing.padding;
```

There might be more places like this.

Reply:
OK, NVM, it seems it's an object...